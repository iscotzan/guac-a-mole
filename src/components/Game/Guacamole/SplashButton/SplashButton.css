.splash-button {
  --offset: 10px;
  --border-size: 2px;
  display: block;
  position: relative;
  padding: 0.5em 1em;
  appearance: none;
  border: 0;
  background: rgba(0, 0, 0, 0.3);
  color: #91ff3f;
  text-transform: uppercase;
  letter-spacing: 0.25em;
  outline: none;
  cursor: pointer;
  font-weight: bold;
  border-radius: 0;
  box-shadow: inset 0 0 0 var(--border-size) currentcolor;
  transition: background 0.8s ease; }
  .splash-button:hover {
    background: rgba(0, 0, 0, 0.45); }
  .splash-button__horizontal, .splash-button__vertical {
    position: absolute;
    top: var(--horizontal-offset, 0);
    right: var(--vertical-offset, 0);
    bottom: var(--horizontal-offset, 0);
    left: var(--vertical-offset, 0);
    transition: transform 0.8s ease;
    will-change: transform; }
    .splash-button__horizontal::before, .splash-button__vertical::before {
      content: "";
      position: absolute;
      border: inherit; }
  .splash-button__horizontal {
    --vertical-offset: calc(var(--offset) * -1);
    border-top: var(--border-size) solid currentcolor;
    border-bottom: var(--border-size) solid currentcolor; }
    .splash-button__horizontal::before {
      top: calc(var(--vertical-offset) - var(--border-size));
      bottom: calc(var(--vertical-offset) - var(--border-size));
      left: calc(var(--vertical-offset) * -1);
      right: calc(var(--vertical-offset) * -1); }
  .splash-button:hover .splash-button__horizontal {
    transform: scaleX(0); }
  .splash-button__vertical {
    --horizontal-offset: calc(var(--offset) * -1);
    border-left: var(--border-size) solid currentcolor;
    border-right: var(--border-size) solid currentcolor; }
    .splash-button__vertical::before {
      top: calc(var(--horizontal-offset) * -1);
      bottom: calc(var(--horizontal-offset) * -1);
      left: calc(var(--horizontal-offset) - var(--border-size));
      right: calc(var(--horizontal-offset) - var(--border-size)); }
  .splash-button:hover .splash-button__vertical {
    transform: scaleY(0); }
