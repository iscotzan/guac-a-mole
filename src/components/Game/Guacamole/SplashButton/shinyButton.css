.draw-border {
    box-shadow: inset 0 0 0 4px #58afd1;
    color: #58afd1;
    transition: color 0.25s 0.08333s;
    position: relative; }
    .draw-border::before, .draw-border::after {
      border: 0 solid transparent;
      box-sizing: border-box;
      content: '';
      pointer-events: none;
      position: absolute;
      width: 0;
      height: 0;
      bottom: 0;
      right: 0; }
    .draw-border::before {
      border-bottom-width: 4px;
      border-left-width: 4px; }
    .draw-border::after {
      border-top-width: 4px;
      border-right-width: 4px; }
    .draw-border:hover {
      color: #ffe593; }
      .draw-border:hover::before, .draw-border:hover::after {
        border-color: #ffe593;
        transition: border-color 0s, width 0.25s, height 0.25s;
        width: 100%;
        height: 100%; }
      .draw-border:hover::before {
        transition-delay: 0s, 0s, 0.25s; }
      .draw-border:hover::after {
        transition-delay: 0s, 0.25s, 0s; }
  
  .btn {
    background: none;
    border: none;
    cursor: pointer;
    line-height: 1.5;
    font: 700 1.2rem 'Roboto Slab', sans-serif;
    padding: 1em 2em;
    letter-spacing: 0.05rem; }
    .btn:focus {
      outline: none;
      /* outline: 2px dotted #55d7dc;  */
    }
  
.shiny-btn {
    color: #E9E45F;
    font-weight: bold;
    font-size: 2rem;
    text-decoration: none;
    text-transform: uppercase;
    /* max-width: 100vw; */
    padding: 40px 90px;
    background-color: rgba(0,0,0,0.5);
    transition: transform 500ms cubic-bezier(0.68, -0.55, 0.265, 1.55), background-position 800ms cubic-bezier(0.68, -0.55, 0.265, 1.55), box-shadow 500ms linear;
    background-image: url('data:image/svg+xml;utf8,<?xml version="1.0" encoding="utf-8"?><svg version="1.1" id="Livello_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"   width="1296px" height="768px" viewBox="0 0 1296 768" enable-background="new 0 0 1296 768" xml:space="preserve"><g><polygon fill="%238694D1" points="766.6,1.2 -0.2,768 200.7,768 967.5,1.2   "/></g><g><polygon fill="%238694D1" points="1094.8,1.2 328,768 528.9,768 1295.7,1.2   "/></g></svg>');
    background-size: contain;
    background-position: -200px center;
    background-repeat: no-repeat;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  }
  .shiny-btn:hover {
    transform: scale(0.7);
    background-position: -60px;
    box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
  }
  .shiny-btn:active {
    transform: scale(0.6);
    background-position: 500px;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  }