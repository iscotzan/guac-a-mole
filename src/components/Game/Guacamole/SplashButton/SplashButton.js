import * as React from "react";
import "./SplashButton.css";
const SplashButton = props => {
  return (
    <div>
      <button className="splash-button" onClick={props.onClick}>
        {props.buttonText}
        <div className="splash-button__horizontal" />
        <div className="splash-button__vertical" />
      </button>
    </div>
  );
};

export default SplashButton;
