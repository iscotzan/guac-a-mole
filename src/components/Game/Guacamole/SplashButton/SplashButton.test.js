import React from "react";
import SplashButton from "./SplashButton";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const wrapper = shallow(<SplashButton buttonText={'buttonText'} onClick={() => {}} />);
describe("<SplashButton />", () => {
  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("renders without crashing", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
