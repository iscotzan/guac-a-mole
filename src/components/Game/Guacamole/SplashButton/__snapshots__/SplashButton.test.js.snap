// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SplashButton /> renders without crashing 1`] = `
<div>
  <button
    className="splash-button"
    onClick={[Function]}
  >
    buttonText
    <div
      className="splash-button__horizontal"
    />
    <div
      className="splash-button__vertical"
    />
  </button>
</div>
`;
